tasks.py

```diff
@@ -1,34 +1,93 @@
+import os
 from pathlib import Path
+from shutil import which
 
 from invoke import task
 
-PKG_NAME = 'more_categories'
-PKG_PATH = Path(f'pelican/plugins/{PKG_NAME}')
-FLAKE8 = 'flake8'
-ISORT = 'isort'
-PYTEST = 'pytest'
+PKG_NAME = "more_categories"
+PKG_PATH = Path(f"pelican/plugins/{PKG_NAME}")
+
+ACTIVE_VENV = os.environ.get("VIRTUAL_ENV", None)
+VENV_HOME = Path(os.environ.get("WORKON_HOME", "~/.local/share/virtualenvs"))
+VENV_PATH = Path(ACTIVE_VENV) if ACTIVE_VENV else (VENV_HOME / PKG_NAME)
+VENV = str(VENV_PATH.expanduser())
+
+TOOLS = ["poetry", "pre-commit"]
+POETRY = which("poetry") if which("poetry") else (VENV / Path("bin") / "poetry")
+PRECOMMIT = (
+    which("pre-commit") if which("pre-commit") else (VENV / Path("bin") / "pre-commit")
+)
+
+LINT_TOOLS = ["flake8", "isort", "black"]
 
 
 @task
 def tests(c):
     """Run the test suite"""
-    c.run(f'{PYTEST}', pty=True)
+    c.run(f"{VENV}/bin/pytest", pty=True)
 
 
 @task
-def isort(c, check=False):
-    check_flag = ''
+def black(c, check=False, diff=False):
+    """Run Black auto-formatter, optionally with --check or --diff"""
+    check_flag, diff_flag = "", ""
     if check:
-        check_flag = '-c'
-    c.run(f'{ISORT} {check_flag} --recursive {PKG_PATH}/* tasks.py')
+        check_flag = "--check"
+    if diff:
+        diff_flag = "--diff"
+    c.run(f"{VENV}/bin/black {check_flag} {diff_flag} {PKG_PATH} tasks.py")
```

Comment:
This repo doesn't use black.

Reply:
All the other ones do... It would make sense to have a unified coding standard acorss the board no?

Reply:
Since plugin authors are volunteering their time and effort to maintain plugins, it's best to honor their wishes in this regard. For background, please read: https://github.com/getpelican/cookiecutter-pelican-plugin/issues/2

Reply:
I'll go ahead and undo this.